Xamarin.Forms.Core/CornerRadius.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Xamarin.Forms.Core
+{
+	public struct CornerRadius
+	{
+		public double TopLeft { get; set; }
+		public double TopRight { get; set; }
+		public double BottomLeft { get; set; }
+		public double BottomRight { get; set; }
+
+		public CornerRadius(double uniformRadius) : this(uniformRadius, uniformRadius, uniformRadius, uniformRadius)
+		{
+		}
+
+		public CornerRadius(double topLeft, double topRight, double bottomLeft, double bottomRight)
+		{
+			if (topLeft < 0 || double.IsNaN(topLeft))
+				throw new ArgumentException("TopLeft is less than 0 or is not a number", "topLeft");
```

Comment:
Can we use `nameof` for the `ArgumentException` messages here?

Reply:
Absolutely!